documentation/source/coroutines.rst

```diff
@@ -151,3 +151,35 @@ they'd naturally end.
         if not isclose(edge_time_ns, start_time_ns + 4.0):
             raise TestFailure("Expected a period of 4 ns")
 
+
+Async functions
+---------------
+
+Python 3.5 introduces ``async`` functions, which provide an alternative syntax.
+For example:
+
+.. code-block:: python
+
+    @cocotb.coroutine
+    async def wait_10ns():
+        cocotb.log.info("About to wait for 10ns")
+        await Timer(10000)
```

Comment:
I'd use ``Timer(10, units='ns')`` for this.

Reply:
It's an example code used earlier in the docs as well, and yes, your option is nicer to read. So if you change it, please change both examples.

Reply:
Leaving for future cleanup in another PR

Reply:
opened https://github.com/potentialventures/cocotb/issues/868